DSPy auto tracing (#13543)
mlflow/mlflow: docs/source/llms/tracing/index.rst

MLflow offers a number of different options to enable tracing of your GenAI applications. 

> - **Automated tracing**: MLflow provides a fully automated integration with various GenAI libraries such as LangChain, OpenAI, LlamaIndex, DSPy, AutoGen, and more, that can activate by simply enabling ``mlflow.<library>.autolog()``.

Reviewer: ```suggestion
- **Automated tracing**: MLflow provides fully automated integrations with various GenAI libraries such as LangChain, OpenAI, LlamaIndex, DSPy, AutoGen, and more that can be activated by simply enabling ``mlflow.<library>.autolog()``.
```

            :width: 100%
            :align: center

    .. tab:: DSPy

        .. raw:: html

            <h3>DSPy Automatic Tracing</h3>

        |

        The MLflow DSPy flavor's autologging feature has a direct integration with MLflow tracing. When DSPy autologging is enabled with :py:func:`mlflow.dspy.autolog`, invocation of components
        such as LMs, Adapters, Modules, will automatically record generated traces during interactive development.

        .. code-block:: python

            import mlflow
            import dspy

            # Enable tracing for DSPy
            mlflow.dspy.autolog()

            # Set an experiment to log the traces to
            mlflow.set_experiment("DSPy Tracing")

            # Define a simple ChainOfThought model and run it
            lm = dspy.LM("openai/gpt-4o-mini")
            dspy.configure(lm=lm)

>             cot = dspy.ChainOfThought("question -> response")

Author: I actually got this from the quickstart😅 Is there any doc/reference for the new standard pattern?

            :width: 100%
            :align: center

    .. tab:: DSPy

        .. raw:: html

            <h3>DSPy Automatic Tracing</h3>

        |

        The MLflow DSPy flavor's autologging feature has a direct integration with MLflow tracing. When DSPy autologging is enabled with :py:func:`mlflow.dspy.autolog`, invocation of components

>         such as LMs, Adapters, Modules, will automatically record generated traces during interactive development.

Reviewer: ```suggestion
        such as LMs, Adapters and Modules will automatically record generated traces during interactive development.
```

        .. code-block:: python

            import mlflow
            import dspy

            # Enable tracing for DSPy
            mlflow.dspy.autolog()

            # Set an experiment to log the traces to
            mlflow.set_experiment("DSPy Tracing")

            # Define a simple ChainOfThought model and run it
            lm = dspy.LM("openai/gpt-4o-mini")
            dspy.configure(lm=lm)

            # Define a simple summarizer model and run it
            class SummarizeSignature(dspy.Signature):
                """Given a passage, generate a summary."""

>                 passage = dspy.InputField(desc="a passage to summarize")

Reviewer 2: ```suggestion
                passage: str = dspy.InputField(desc="a passage to summarize")
```

nit
